src/main/java/org/gridsuite/network/map/NetworkMapService.java

```diff
@@ -1140,7 +1202,48 @@ public List<LineMapData> getMapLines(UUID networkUuid, String variantId, List<St
                     network.getSubstation(id).getVoltageLevelStream().forEach(v ->
                             v.getConnectables(Line.class).forEach(l -> lines.add(toBasicMapData(l)))));
             return new ArrayList<>(lines);
+        }
+    }
+
+    public List<String> getHvdcLinesIds(UUID networkUuid, String variantId, List<String> substationsIds) {
+        Network network = getNetwork(networkUuid, getPreloadingStrategy(substationsIds), variantId);
+        if (substationsIds == null) {
+            return network.getHvdcLineStream()
+                    .map(HvdcLine::getId).collect(Collectors.toList());
+        } else {
+            return substationsIds.stream()
+                    .flatMap(substationsId -> network.getSubstation(substationsId).getVoltageLevelStream())
+                    .flatMap(voltageLevel -> voltageLevel.getConnectableStream(HvdcConverterStation.class))
+                    .filter(hvdcConverterStation -> hvdcConverterStation.getHvdcLine() != null)
+                    .map(HvdcConverterStation::getHvdcLine)
```

Comment:
you can filter after getting the hvdcLine
.map(HvdcConverterStation::getHvdcLine)
.filter(Objects::nonNull)